src/AdvApi32.Desktop/AdvApi32.cs

```diff
@@ -0,0 +1,340 @@
+            SC_MANAGER_ENUMERATE_SERVICE = 0x0004,
+            SC_MANAGER_LOCK = 0x0008,
+            SC_MANAGER_QUERY_LOCK_STATUS = 0x0010,
+            SC_MANAGER_MODIFY_BOOT_CONFIG = 0x0020,
+            SC_MANAGER_ALL_ACCESS = STANDARD_RIGHTS_REQUIRED |
+                                        SC_MANAGER_CONNECT |
+                                        SC_MANAGER_CREATE_SERVICE |
+                                        SC_MANAGER_ENUMERATE_SERVICE |
+                                        SC_MANAGER_LOCK |
+                                        SC_MANAGER_QUERY_LOCK_STATUS |
+                                        SC_MANAGER_MODIFY_BOOT_CONFIG
+        }
+
+        /// <summary>
+        /// Describes service access flags.
+        /// </summary>
+        [Flags]
+        public enum ServiceAccess : uint
+        {
+            GenericRead = 0x80000000,
+            GenericWrite = 0x40000000,
+            GenericExecute = 0x20000000,
+
+            AccessSystemSecurity = 0x1000000,
+            Delete = 0x10000,
+            ReadControl = 0x20000,
+            WriteDAC = 0x40000,
+            WriteOwner = 0x80000,
+
+            STANDARD_RIGHTS_REQUIRED = 0xF0000,
+
+            SERVICE_QUERY_CONFIG = 0x0001,
+            SERVICE_CHANGE_CONFIG = 0x0002,
+            SERVICE_QUERY_STATUS = 0x0004,
+            SERVICE_ENUMERATE_DEPENDENTS = 0x0008,
+            SERVICE_START = 0x0010,
+            SERVICE_STOP = 0x0020,
+            SERVICE_PAUSE_CONTINUE = 0x0040,
+            SERVICE_INTERROGATE = 0x0080,
+            SERVICE_USER_DEFINED_CONTROL = 0x0100,
+            SERVICE_ALL_ACCESS = STANDARD_RIGHTS_REQUIRED |
+                                        SERVICE_QUERY_CONFIG |
+                                        SERVICE_CHANGE_CONFIG |
+                                        SERVICE_QUERY_STATUS |
+                                        SERVICE_ENUMERATE_DEPENDENTS |
+                                        SERVICE_START |
+                                        SERVICE_STOP |
+                                        SERVICE_PAUSE_CONTINUE |
+                                        SERVICE_INTERROGATE |
+                                        SERVICE_USER_DEFINED_CONTROL
+        }
+
+        /// <summary>
+        /// Describes service type flags.
+        /// </summary>
+        [Flags]
+        public enum ServiceType : uint
+        {
+            SERVICE_KERNEL_DRIVER = 0x00000001,
+            SERVICE_FILE_SYSTEM_DRIVER = 0x00000002,
+            SERVICE_ADAPTER = 0x00000004,
+            SERVICE_RECOGNIZER_DRIVER = 0x00000008,
+            SERVICE_WIN32_OWN_PROCESS = 0x00000010,
+            SERVICE_WIN32_SHARE_PROCESS = 0x00000020,
+            SERVICE_INTERACTIVE_PROCESS = 0x00000100
+        }
+
+        /// <summary>
+        /// Describes service start type.
+        /// </summary>
+        public enum ServiceStartType : uint
+        {
+            SERVICE_BOOT_START,
```

Comment:
Are you confident that the implied values generated by C# for these enums match those in the native headers?